### DISCUSSION TITLE
Backup restic backup repo outside fly.io

### DISCUSSION THREAD

**Discussion by lp-ch** [arthurgeek/vaultwarden-fly-template#33] [Ideas]
Hi,

First of all, I would like to thank you very much for your work! I was able to run my own vaultwarden serven on fly.io. It would have been certainly much harder, or even impossible for me, without you.

I am now concerned about an automated backup outside fly.io. Because I am running my instance on a free tier, if the volume crash or data become unavailable, vaultwarden is gone.

restic backup is stored in repo on the same volume. I am not feeling really secure with that.

For pulling data outside fly.io, I though about WebDAV or S3 storage. Regarding S3 storage, [MinIO](https://min.io/docs/minio/container/index.html) seems to be an option. Synology CloudSync can easily connect to booth. WebDAV is not so standard in backup tools, but S3 support certainly is. 

For pushing data outside fly.io, I read an article about [automating Fly.io database Backups with GitHub Actions](https://rodolfosilva.com/github-actions/automating-flyio-database-backups-with-github-actions). It would push the backup to a (private) GitHub repo.

I am currently not enough skilled to implement any of these option. If somebody would like implement it in this vaultwarden-fly-template, I would really enjoy to use it ;)!

What do you think about this idea?

**Comment by arthurgeek** [comment]
Hi @lp-ch!

Thanks for the kind words, appreciate it.

Today is already possible to backup to S3 or any S3-based choices, or [any other repository supported by Restic](https://restic.readthedocs.io/en/stable/030_preparing_a_new_repo.html).

All you need to do is to add the following variables here: https://github.com/arthurgeek/vaultwarden-fly-template/blob/443df1f3689a8737b99c505437fbb48b7b5b4684/.config.sample.env#L35-L47

`RESTIC_REPOSITORY` is your URL, in the S3 case, it would be something like: http://s3.amazonaws.com/[bucket_name]/. It can also be a Minio, R2, Wasabi, and others URL.

`RESTIC_PASSWORD` is a password of your choosing, used to encrypt the backup.

Finally, you need `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.

After adding those, you need to follow the 2nd FAQ : https://github.com/arthurgeek/vaultwarden-fly-template?tab=readme-ov-file#faq.

Let me know if it works for you!
